deepchecks/nlp/checks/train_test_validation/keyword_frequency_drift.py

```diff
@@ -0,0 +1,142 @@
+from deepchecks.nlp.base_checks import TrainTestCheck
+from deepchecks.utils.distribution.drift import cramers_v, psi
+from typing import Union, List, Any
+from deepchecks.nlp.context import Context
+from sklearn.feature_extraction.text import TfidfVectorizer
+from nltk.stem.porter import PorterStemmer
+import numpy as np
+
+__all__ = ['KeywordFrequencyDrift']
+
+
+class KeywordFrequencyDrift(TrainTestCheck):
+    """
+    Computes the keywords' frequencies drift between the train and the test datasets.
+
+    Parameters
+    ----------
+    top_n_to_show: int, default: 20
+        How many words will be displayed in the graph.
+    top_n_method: Union[str, List[str]], default: 'top_diff'
+        Decides which method will be used to select the top n words to show. Possible values are:
+        "top_diff" for the top difference between train and test, "top_freq" for the top absolute frequencies, or a list
+        of keywords in which case the words on the list are shown and the "top_n_to_show" parameter is ignored.
+    drift_method: str, default: "cramer_v"
+        Decides which method will be used for drift calculation. Possible values are:
+        "cramer_v" for Cramer's V, "PSI" for Population Stability Index (PSI).
+    """
+
+    def __init__(self,
+                 top_n_to_show: int = 20,
+                 top_n_method: Union[str, List[str]] = 'top_diff',
+                 drift_method='cramer_v',
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.top_n_to_show = top_n_to_show
+        self.top_n_method = top_n_method
+
+        if drift_method == 'PSI':
+            self.drift_method = psi
+        elif drift_method == 'cramer_v':
+            self.drift_method = cramers_v
+        else:
+            raise DeepchecksValueError('drift_method must be one of: PSI, cramer_v')
+        self.stem_func = PorterStemmer().stem
+        self.token_pattern = r'[a-z]{2,}'
+
+        self.top_n_words = None
+        self.top_n_diffs = None
+
+    def run_logic(self, context: Context) -> CheckResult:
+        """Run check."""
+        train_dataset = context.train
+        test_dataset = context.test
+        all_data = list(train_dataset.text) + list(test_dataset.text)
+
+        vectorizer = TfidfVectorizer(input='content', strip_accents='ascii', preprocessor=self.stem_func,
+                                     token_pattern=self.token_pattern)
+        vectorizer.fit(all_data)
+        train_freqs = vectorizer.transform(train_dataset.text)
+        max_train_freqs = np.array(train_freqs.max(axis=0).todense()).reshape(-1)
+
+        test_freqs = vectorizer.transform(test_dataset.text)
+        max_test_freqs = np.array(test_freqs.max(axis=0).todense()).reshape(-1)
+        word_freq_diff = np.abs(max_train_freqs - max_test_freqs)
+
+        max_train_counts = max_train_freqs * test_dataset.n_samples
+        max_test_counts = max_test_freqs * test_dataset.n_samples
```

Comment:
This promises a scale of 0 to 1?

Reply:
The opposite - this makes it in total counts instead of fraction, the format that both Cramer's V and PSI expect.

Reply:
Oh right, and the plot is of the version before multiplying by n_samples?

Reply:
correct